bot/src/main/java/com/dongtronic/diabot/platforms/discord/commands/quote/QuoteSearchCommand.kt

```diff
@@ -0,0 +1,119 @@
+package com.dongtronic.diabot.platforms.discord.commands.quote
+
+import com.dongtronic.diabot.util.Patterns
+import com.dongtronic.diabot.data.mongodb.QuoteDAO
+import com.dongtronic.diabot.data.mongodb.QuoteDTO
+import com.dongtronic.diabot.platforms.discord.commands.DiscordCommand
+import com.jagrosh.jdautilities.command.CommandEvent
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.launch
+import org.litote.kmongo.regex
+import org.litote.kmongo.and
+import org.bson.conversions.Bson
+import net.dv8tion.jda.api.EmbedBuilder
+
+class QuoteSearchCommand(category: Category, parent: QuoteCommand) : DiscordCommand(category, parent) {
+    private val maxNumberOfQuotes: Int
+
+    init {
+        this.name = "search"
+        this.help = "Searches through quotes for keywords (quotes contain *all* key words)"
+        this.guildOnly = true
+        this.aliases = arrayOf("s")
+        this.examples = arrayOf(
+            "diabot quote search some funny words",
+            "diabot quote search r keyword"
+        )
+
+        this.maxNumberOfQuotes = System.getenv().getOrDefault("QUOTE_MAX_SEARCH_DISPLAY", "10").toInt()
+    }
+
+    override fun execute(event: CommandEvent) {
+        runBlocking {
+            launch {
+                if (!QuoteDAO.awaitCheckRestrictions(event.guildChannel, warnDisabledGuild = true)) return@launch
+
+                var args = event.message.contentRaw.split(" ").toList()
+                if (args.size < 3) {
+                    replyTooFewArgs(event)
+                    return@launch
+                }
+
+                // Remove the command portion of the message
+                args = args.slice(2..args.size - 1)
+
+                // Do we want one random quote from our search?
+                var random = false
+                if (args[0] == "r") {
+                    random = true
+                    if (args.size < 2) {
+                        replyTooFewArgs(event)
+                        return@launch
+                    }
+
+                    args = args.slice(1..args.size - 1)
+                }
+
+                var keywords: List<String> = emptyList()
+                for (arg in args) {
+                    keywords += sanitizeString(arg)
+                }
+
+                // Build a list of regex filters based on the provided keywords
+                var filters: List<Bson> = emptyList()
+                for (kw in keywords) {
+                    filters += QuoteDTO::message regex "(?i)$kw"
+                }
+
+                @Suppress("SwallowedException")
```

Comment:
Is this still needed? When I remove this I don't see any warnings on my end.

Reply:
My build setup is using detekt. Maybe with the wrong config? Idk, I don't think my build setup is correct, I had to modify the dev docker image to use a newer gradle image, not sure what's supposed to work.

I'll just disable the detekt check and push, I don't know what linting is supposed to be done. The changes from the animal images also breaks detekt, had to tweak some settings about maximum number of functions per file.

Reply:
Oh, if it's detekt that's producing the warning then ignore me. I didn't try running detekt after removing the `@Suppress`, I was just relying on the built-in IntelliJ warnings 😬